File: tests/rust-fixtures/basic/basic.rs
// Test fixture: basic.rs - minimal compilation unit for Rust adapter testing
// @verify chimera-rust-schema
// @expected ch_status SUCCESS

pub struct Point {
    pub x: i32,
    pub y: i32,
}

impl Point {
    pub fn new(x: i32, y: i32) -> Self {
        Point { x, y }
    }

    pub fn distance_squared(&self, other: &Point) -> u64 {
        let dx = self.x - other.x;
        let dy = self.y - other.y;
        ((dx * dx) + (dy * dy)) as u64
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_point_distance() {
        let p1 = Point::new(0, 0);
        let p2 = Point::new(3, 4);
        assert_eq!(p1.distance_squared(&p2), 25);
    }
}

File: tests/c-fixtures/basic/basic.c
#include "basic.h"

int add(int a, int b) { return a + b; }

int multiply(int a, int b) { return a * b; }

int point_distance(struct Point p) {
    int x = p.x < 0 ? -p.x : p.x;
    int y = p.y < 0 ? -p.y : p.y;
    return x + y;
}
